src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/scatter_update/scatter_nd_update_kernel_ref.cpp

```diff
@@ -115,13 +113,10 @@ bool ScatterNDUpdateKernelRef::Validate(const Params& p, const optional_params&
     return true;
 }
 
-static std::string GetInputBlockND(const scatter_nd_update_params& params) {
-    const auto& input = params.inputs[0];
+static std::string GetInputBlockND(const scatter_nd_update_params& params, int num) {
+    const auto& input = params.inputs[num];
     auto input_dims = input.LogicalDims();
     std::reverse(input_dims.begin(), input_dims.end());
-    while (!input_dims.empty() && input_dims.back() == 1) {
-        input_dims.pop_back();
-    }
     const int rank = static_cast<int>(input_dims.size());
```

Comment:
Is this okay to be removed?

Reply:
Yes, this was intentional as the exact shape (including where shape size is 1) is needed for the cl kernel